In [1]:
!pip install datasets

INFO: pip is looking at multiple versions of multiprocess to determine which version is compatible with other requirements. This could take a while.
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 471.6/471.6 kB 15.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 116.3/116.3 kB 13.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 134.8/134.8 kB 15.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 194.1/194.1 kB 21.1 MB/s eta 0:00:00


In [2]:
from huggingface_hub import login
from google.colab import userdata


login(userdata.get('HF_TOKEN'))

The token has not been saved to the git credentials helper. Pass `add_to_git_credential=True` in this function directly or `--add-to-git-credential` if using via `huggingface-cli` if you want to set the git credential as well.
Token is valid (permission: read).
Your token has been saved to /root/.cache/huggingface/token
Login successful


In [6]:
from transformers import AutoTokenizer, AutoModelForCausalLM





tokenizer = AutoTokenizer.from_pretrained("google/gemma-2b")
model = AutoModelForCausalLM.from_pretrained("google/gemma-2b", device_map="auto")

tokenizer_config.json:   0%|          | 0.00/33.6k [00:00<?, ?B/s]

tokenizer.model:   0%|          | 0.00/4.24M [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/17.5M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/636 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/627 [00:00<?, ?B/s]

model.safetensors.index.json:   0%|          | 0.00/13.5k [00:00<?, ?B/s]

model-00001-of-00002.safetensors:   0%|          | 0.00/4.95G [00:00<?, ?B/s]

model-00002-of-00002.safetensors:   0%|          | 0.00/67.1M [00:00<?, ?B/s]

`config.hidden_act` is ignored, you should use `config.hidden_activation` instead.
Gemma's activation function will be set to `gelu_pytorch_tanh`. Please, use
`config.hidden_activation` if you want to override this behaviour.
See https://github.com/huggingface/transformers/pull/29402 for more details.


Loading checkpoint shards:   0%|          | 0/2 [00:00<?, ?it/s]

generation_config.json:   0%|          | 0.00/137 [00:00<?, ?B/s]

In [7]:
input_text = "What is your name?"
input_ids = tokenizer(input_text, return_tensors="pt").to("cuda")
outputs = model.generate(**input_ids)
print(tokenizer.decode(outputs[0]))

/usr/local/lib/python3.10/dist-packages/transformers/generation/utils.py:1258: UserWarning: Using the model-agnostic default `max_length` (=20) to control the generation length. We recommend setting `max_new_tokens` to control the maximum length of the generation.
  warnings.warn(


<bos>What is your name?

What is your age?

What is your gender?

What


In [9]:
tokens = input_ids['input_ids']
print(tokens)

logits = model(**input_ids).logits
for i in range(tokens.shape[-1]):
    token = tokens[0, i].item()
    print(logits[0, i, token])

tensor([[     2,   1841,    603,    861,   1503, 235336]], device='cuda:0')
tensor(-18.2747, device='cuda:0', grad_fn=<SelectBackward0>)
tensor(-33.2665, device='cuda:0', grad_fn=<SelectBackward0>)
tensor(-23.9536, device='cuda:0', grad_fn=<SelectBackward0>)
tensor(-27.7627, device='cuda:0', grad_fn=<SelectBackward0>)
tensor(-19.6064, device='cuda:0', grad_fn=<SelectBackward0>)
tensor(-21.0372, device='cuda:0', grad_fn=<SelectBackward0>)


In [10]:
import torch

def zero_shot_classification(text, task_description, labels):  # text는 주어진 입력, task_description은 task에 대한 설명, labels은 class들을 text로 변환한 결과입니다.
    text_ids = tokenizer(task_description + text, return_tensors="pt").to("cuda")  # 먼저 task_description과 text를 이어붙인 후, tokenize합니다.
    probs = []
    for label in labels:  # 그 다음 각 text화된 label들을 tokenize하고 입력에 이어붙인 후, Gemma-2B에 넣어줍니다.
        label_ids = tokenizer(label, return_tensors="pt").to("cuda")
        n_label_tokens = label_ids['input_ids'].shape[-1] - 1  # text로 변환한 label의 token 수를 계산합니다.
        input_ids = {
            'input_ids': torch.concatenate([text_ids['input_ids'], label_ids['input_ids'][:, 1:]], axis=-1),  # concatenate 명령어를 통해 이어붙이는 모습입니다.
            'attention_mask': torch.concatenate([text_ids['attention_mask'], label_ids['attention_mask'][:, 1:]], axis=-1)
        }

        logits = model(**input_ids).logits  # Logit을 계산한 모습입니다.
        prob = 0
        n_total = input_ids['input_ids'].shape[-1]
        for i in range(n_label_tokens, 0, -1):  # 일반적으로 text로 변환한 label은 여러 token으로 이루어져있습니다. 이러한 label에 대한 logit은 구성하는 모든 token들의 logit들의 합으로 정의합니다.
            token = label_ids['input_ids'][0, i].item()
            prob += logits[0, n_total - i, token].item()
        probs.append(prob)

        del input_ids
        del logits
        torch.cuda.empty_cache()  # 위의 del과 empty_cache() 명령어를 통해 GPU를 제때 할당해제 해줍니다. 만약 GPU가 여유롭다면 지워주시는게 속도적으로 이득입니다.

    return probs

In [11]:
probs = zero_shot_classification("I am happy!", "Is the sentence positive or negative?: ", ["positive", "negative"])
print(probs)

[-4.5151824951171875, -9.59005069732666]


In [12]:
from datasets import load_dataset


imdb = load_dataset("imdb")
def preprocess_function(examples):
    return tokenizer(examples["text"], max_length=200, truncation=True)

tokenized_imdb = imdb.map(preprocess_function, batched=True)

README.md:   0%|          | 0.00/7.81k [00:00<?, ?B/s]

train-00000-of-00001.parquet:   0%|          | 0.00/21.0M [00:00<?, ?B/s]

test-00000-of-00001.parquet:   0%|          | 0.00/20.5M [00:00<?, ?B/s]

unsupervised-00000-of-00001.parquet:   0%|          | 0.00/42.0M [00:00<?, ?B/s]

Generating train split:   0%|          | 0/25000 [00:00<?, ? examples/s]

Generating test split:   0%|          | 0/25000 [00:00<?, ? examples/s]

Generating unsupervised split:   0%|          | 0/50000 [00:00<?, ? examples/s]

Map:   0%|          | 0/25000 [00:00<?, ? examples/s]

Map:   0%|          | 0/25000 [00:00<?, ? examples/s]

Map:   0%|          | 0/50000 [00:00<?, ? examples/s]

In [15]:
import numpy as np
from tqdm import tqdm


n_corrects = 0
for i in tqdm(range(50)):
    text = tokenized_imdb['test'][i]['text']
    label = tokenized_imdb['test'][i]['label']
    probs = zero_shot_classification(
        text,
        "A movie review is given. Decide that the movie review is positive or negative: ",
        labels=["Answer: negative.", "Answer: positive."]
    )

    pred = np.argmax(np.array(probs))
    if pred == label:
        n_corrects += 1

print(n_corrects)

100%|██████████| 50/50 [00:52<00:00,  1.06s/it]

44


##과제
이번 과제에서는 이전 주차 과제에서 활용했던 `fancyzhx/ag_news` 문제를 zero-shot classification으로 푸시면 됩니다. 아래 사항들에 유의하시면 될 것 같습니다.

- [ ]  Label들을 올바르게 text화 하여 넘겨주셔야 합니다.
- [ ]  `test` split data 50개에 대한 정확도 계산 코드 및 출력이 남아있어야 합니다.

이외에는 Gemma-2B 모델의 logit 계산 능력을 활용한다는 부분 빼고는 제약이 없습니다.


In [18]:
ag_news = load_dataset("fancyzhx/ag_news")

tokenized_ag_news = ag_news.map(preprocess_function, batched=True)

Map:   0%|          | 0/120000 [00:00<?, ? examples/s]

Map:   0%|          | 0/7600 [00:00<?, ? examples/s]

In [23]:
n_corrects = 0
for i in tqdm(range(20)):
    text = tokenized_ag_news['test'][i]['text']
    label = tokenized_ag_news['test'][i]['label']
    print("text -> ",text )
    print("label -> ",label )
    probs = zero_shot_classification(
        text,
        "A news is given. Decide that the news is positive or negative: ",
        labels=["Answer: negative.", "Answer: positive."]
    )

    pred = np.argmax(np.array(probs))
    print("pred ==> ", pred)
    if pred == label:
        n_corrects += 1

print(n_corrects)

print("n_corrects ==> " ,n_corrects)

  0%|          | 0/20 [00:00<?, ?it/s]

text ->  Fears for T N pension after talks Unions representing workers at Turner   Newall say they are 'disappointed' after talks with stricken parent firm Federal Mogul.
label ->  2


  5%|▌         | 1/20 [00:00<00:05,  3.43it/s]

pred ==>  0
text ->  The Race is On: Second Private Team Sets Launch Date for Human Spaceflight (SPACE.com) SPACE.com - TORONTO, Canada -- A second\team of rocketeers competing for the  #36;10 million Ansari X Prize, a contest for\privately funded suborbital space flight, has officially announced the first\launch date for its manned rocket.
label ->  3


 10%|█         | 2/20 [00:00<00:06,  2.60it/s]

pred ==>  1
text ->  Ky. Company Wins Grant to Study Peptides (AP) AP - A company founded by a chemistry researcher at the University of Louisville won a grant to develop a method of producing better peptides, which are short chains of amino acids, the building blocks of proteins.
label ->  3


 15%|█▌        | 3/20 [00:01<00:06,  2.49it/s]

pred ==>  0
text ->  Prediction Unit Helps Forecast Wildfires (AP) AP - It's barely dawn when Mike Fitzpatrick starts his shift with a blur of colorful maps, figures and endless charts, but already he knows what the day will bring. Lightning will strike in places he expects. Winds will pick up, moist places will dry and flames will roar.
label ->  3


 20%|██        | 4/20 [00:01<00:06,  2.40it/s]

pred ==>  0
text ->  Calif. Aims to Limit Farm-Related Smog (AP) AP - Southern California's smog-fighting agency went after emissions of the bovine variety Friday, adopting the nation's first rules to reduce air pollution from dairy cow manure.
label ->  3


 25%|██▌       | 5/20 [00:02<00:06,  2.39it/s]

pred ==>  0
text ->  Open Letter Against British Copyright Indoctrination in Schools The British Department for Education and Skills (DfES) recently launched a "Music Manifesto" campaign, with the ostensible intention of educating the next generation of British musicians. Unfortunately, they also teamed up with the music industry (EMI, and various artists) to make this popular. EMI has apparently negotiated their end well, so that children in our schools will now be indoctrinated about the illegality of downloading music.The ignorance and audacity of this got to me a little, so I wrote an open letter to the DfES about it. Unfortunately, it's pedantic, as I suppose you have to be when writing to goverment representatives. But I hope you find it useful, and perhaps feel inspired to do something similar, if or when the same thing has happened in your area.
label ->  3


 30%|███       | 6/20 [00:02<00:07,  1.90it/s]

pred ==>  0
text ->  Loosing the War on Terrorism \\"Sven Jaschan, self-confessed author of the Netsky and Sasser viruses, is\responsible for 70 percent of virus infections in 2004, according to a six-month\virus roundup published Wednesday by antivirus company Sophos."\\"The 18-year-old Jaschan was taken into custody in Germany in May by police who\said he had admitted programming both the Netsky and Sasser worms, something\experts at Microsoft confirmed. (A Microsoft antivirus reward program led to the\teenager's arrest.) During the five months preceding Jaschan's capture, there\were at least 25 variants of Netsky and one of the port-scanning network worm\Sasser."\\"Graham Cluley, senior technology consultant at Sophos, said it was staggeri ...\\
label ->  3


 35%|███▌      | 7/20 [00:03<00:07,  1.63it/s]

pred ==>  0
text ->  FOAFKey: FOAF, PGP, Key Distribution, and Bloom Filters \\FOAF/LOAF  and bloom filters have a lot of interesting properties for social\network and whitelist distribution.\\I think we can go one level higher though and include GPG/OpenPGP key\fingerpring distribution in the FOAF file for simple web-of-trust based key\distribution.\\What if we used FOAF and included the PGP key fingerprint(s) for identities?\This could mean a lot.  You include the PGP key fingerprints within the FOAF\file of your direct friends and then include a bloom filter of the PGP key\fingerprints of your entire whitelist (the source FOAF file would of course need\to be encrypted ).\\Your whitelist would be populated from the social network as your client\discovered new identit ...\\
label ->  3


 40%|████      | 8/20 [00:04<00:07,  1.53it/s]

pred ==>  0
text ->  E-mail scam targets police chief Wiltshire Police warns about "phishing" after its fraud squad chief was targeted.
label ->  3


 45%|████▌     | 9/20 [00:04<00:05,  1.89it/s]

pred ==>  0
text ->  Card fraud unit nets 36,000 cards In its first two years, the UK's dedicated card fraud unit, has recovered 36,000 stolen cards and 171 arrests - and estimates it saved 65m.
label ->  3


 50%|█████     | 10/20 [00:04<00:04,  2.00it/s]

pred ==>  0
text ->  Group to Propose New High-Speed Wireless Format  LOS ANGELES (Reuters) - A group of technology companies  including Texas Instruments Inc. &lt;TXN.N&gt;, STMicroelectronics  &lt;STM.PA&gt; and Broadcom Corp. &lt;BRCM.O&gt;, on Thursday said they  will propose a new wireless networking standard up to 10 times  the speed of the current generation.
label ->  3


 55%|█████▌    | 11/20 [00:05<00:04,  2.03it/s]

pred ==>  0
text ->  Apple Launches Graphics Software, Video Bundle  LOS ANGELES (Reuters) - Apple Computer Inc.&lt;AAPL.O&gt; on  Tuesday began shipping a new program designed to let users  create real-time motion graphics and unveiled a discount  video-editing software bundle featuring its flagship Final Cut  Pro software.
label ->  3


 60%|██████    | 12/20 [00:05<00:03,  2.09it/s]

pred ==>  0
text ->  Dutch Retailer Beats Apple to Local Download Market  AMSTERDAM (Reuters) - Free Record Shop, a Dutch music  retail chain, beat Apple Computer Inc. to market on Tuesday  with the launch of a new download service in Europe's latest  battleground for digital song services.
label ->  3


 65%|██████▌   | 13/20 [00:06<00:03,  2.15it/s]

pred ==>  0
text ->  Super ant colony hits Australia A giant 100km colony of ants  which has been discovered in Melbourne, Australia, could threaten local insect species.
label ->  3


 70%|███████   | 14/20 [00:06<00:02,  2.47it/s]

pred ==>  0
text ->  Socialites unite dolphin groups Dolphin groups, or "pods", rely on socialites to keep them from collapsing, scientists claim.
label ->  3


 75%|███████▌  | 15/20 [00:06<00:01,  2.77it/s]

pred ==>  0
text ->  Teenage T. rex's monster growth Tyrannosaurus rex achieved its massive size due to an enormous growth spurt during its adolescent years.
label ->  3


 80%|████████  | 16/20 [00:07<00:01,  3.01it/s]

pred ==>  0
text ->  Scientists Discover Ganymede has a Lumpy Interior Jet Propulsion Lab -- Scientists have discovered irregular lumps beneath the icy surface of Jupiter's largest moon, Ganymede. These irregular masses may be rock formations, supported by Ganymede's icy shell for billions of years...
label ->  3


 85%|████████▌ | 17/20 [00:07<00:01,  2.73it/s]

pred ==>  0
text ->  Mars Rovers Relay Images Through Mars Express European Space Agency -- ESAs Mars Express has relayed pictures from one of NASA's Mars rovers for the first time, as part of a set of interplanetary networking demonstrations.     The demonstrations pave the way for future Mars missions to draw on joint interplanetary networking capabilities...
label ->  3


 90%|█████████ | 18/20 [00:08<00:00,  2.55it/s]

pred ==>  0
text ->  Rocking the Cradle of Life When did life begin? One evidential clue stems from the fossil records in Western Australia, although whether these layered sediments are biological or chemical has spawned a spirited debate. Oxford researcher, Nicola McLoughlin, describes some of the issues in contention.
label ->  3


 95%|█████████▌| 19/20 [00:08<00:00,  2.45it/s]

pred ==>  0
text ->  Storage, servers bruise HP earnings update Earnings per share rise compared with a year ago, but company misses analysts' expectations by a long shot.
label ->  3


100%|██████████| 20/20 [00:08<00:00,  2.29it/s]

pred ==>  0
0
n_corrects ==>  0


In [22]:
n_corrects = 0
for i in tqdm(range(500)):
    text = tokenized_ag_news['test'][i]['text']
    label = tokenized_ag_news['test'][i]['label']
    probs = zero_shot_classification(
        text,
        "A news is given. Decide that the news is positive or negative: ",
        labels=["Answer: negative.", "Answer: positive."]
    )

    pred = np.argmax(np.array(probs))
    print("pred ==> ", pred)
    if pred == label:
        n_corrects += 1

print(n_corrects)

print("n_corrects ==> " ,n_corrects)

  0%|          | 1/500 [00:00<02:49,  2.95it/s]

pred ==>  0


  0%|          | 2/500 [00:00<03:29,  2.38it/s]

pred ==>  1


  1%|          | 3/500 [00:01<03:35,  2.31it/s]

pred ==>  0


  1%|          | 4/500 [00:01<03:44,  2.21it/s]

pred ==>  0


  1%|          | 5/500 [00:02<03:45,  2.20it/s]

pred ==>  0


  1%|          | 6/500 [00:03<04:44,  1.73it/s]

pred ==>  0


  1%|▏         | 7/500 [00:03<05:30,  1.49it/s]

pred ==>  0


  2%|▏         | 8/500 [00:04<05:45,  1.42it/s]

pred ==>  0


  2%|▏         | 9/500 [00:04<04:37,  1.77it/s]

pred ==>  0


  2%|▏         | 10/500 [00:05<04:26,  1.84it/s]

pred ==>  0


  2%|▏         | 11/500 [00:05<04:24,  1.85it/s]

pred ==>  0


  2%|▏         | 12/500 [00:06<04:20,  1.88it/s]

pred ==>  0


  3%|▎         | 13/500 [00:06<04:10,  1.94it/s]

pred ==>  0


  3%|▎         | 14/500 [00:07<03:43,  2.17it/s]

pred ==>  0


  3%|▎         | 15/500 [00:07<03:23,  2.39it/s]

pred ==>  0


  3%|▎         | 16/500 [00:07<03:09,  2.55it/s]

pred ==>  0


  3%|▎         | 17/500 [00:08<03:25,  2.35it/s]

pred ==>  0


  4%|▎         | 18/500 [00:08<03:31,  2.28it/s]

pred ==>  0


  4%|▍         | 19/500 [00:09<03:37,  2.22it/s]

pred ==>  0


  4%|▍         | 20/500 [00:09<03:08,  2.54it/s]

pred ==>  0


  4%|▍         | 21/500 [00:09<02:49,  2.82it/s]

pred ==>  0


  4%|▍         | 22/500 [00:10<02:35,  3.07it/s]

pred ==>  0


  5%|▍         | 23/500 [00:10<02:25,  3.28it/s]

pred ==>  0


  5%|▍         | 24/500 [00:10<02:48,  2.83it/s]

pred ==>  0


  5%|▌         | 25/500 [00:11<02:59,  2.64it/s]

pred ==>  0


  5%|▌         | 26/500 [00:11<02:43,  2.90it/s]

pred ==>  0


  5%|▌         | 27/500 [00:11<02:33,  3.09it/s]

pred ==>  0


  6%|▌         | 28/500 [00:12<02:52,  2.74it/s]

pred ==>  0


  6%|▌         | 29/500 [00:12<03:05,  2.54it/s]

pred ==>  0


  6%|▌         | 30/500 [00:13<03:16,  2.39it/s]

pred ==>  0


  6%|▌         | 31/500 [00:13<03:19,  2.36it/s]

pred ==>  0


  6%|▋         | 32/500 [00:13<02:58,  2.62it/s]

pred ==>  0


  7%|▋         | 33/500 [00:14<03:07,  2.49it/s]

pred ==>  0


  7%|▋         | 34/500 [00:14<03:16,  2.37it/s]

pred ==>  0


  7%|▋         | 35/500 [00:15<03:22,  2.29it/s]

pred ==>  0


  7%|▋         | 36/500 [00:15<03:27,  2.24it/s]

pred ==>  0


  7%|▋         | 37/500 [00:16<03:00,  2.56it/s]

pred ==>  0


  8%|▊         | 38/500 [00:16<03:09,  2.44it/s]

pred ==>  0


  8%|▊         | 39/500 [00:16<03:14,  2.37it/s]

pred ==>  0


  8%|▊         | 40/500 [00:17<03:16,  2.34it/s]

pred ==>  0


  8%|▊         | 41/500 [00:17<03:18,  2.31it/s]

pred ==>  0


  8%|▊         | 42/500 [00:18<03:21,  2.27it/s]

pred ==>  0


  9%|▊         | 43/500 [00:18<03:22,  2.26it/s]

pred ==>  0


  9%|▉         | 44/500 [00:19<03:26,  2.21it/s]

pred ==>  0


  9%|▉         | 45/500 [00:19<03:29,  2.18it/s]

pred ==>  0


  9%|▉         | 46/500 [00:20<03:05,  2.45it/s]

pred ==>  0


  9%|▉         | 47/500 [00:20<03:12,  2.35it/s]

pred ==>  0


 10%|▉         | 48/500 [00:20<03:18,  2.27it/s]

pred ==>  0


 10%|▉         | 49/500 [00:21<03:20,  2.25it/s]

pred ==>  0


 10%|█         | 50/500 [00:21<03:19,  2.25it/s]

pred ==>  0


 10%|█         | 51/500 [00:22<03:19,  2.25it/s]

pred ==>  0


 10%|█         | 52/500 [00:22<03:22,  2.21it/s]

pred ==>  1


 11%|█         | 53/500 [00:23<03:19,  2.24it/s]

pred ==>  0


 11%|█         | 54/500 [00:23<02:58,  2.50it/s]

pred ==>  0


 11%|█         | 55/500 [00:23<03:07,  2.37it/s]

pred ==>  0


 11%|█         | 56/500 [00:24<02:46,  2.67it/s]

pred ==>  0


 11%|█▏        | 57/500 [00:24<02:56,  2.50it/s]

pred ==>  0


 12%|█▏        | 58/500 [00:25<03:04,  2.39it/s]

pred ==>  0


 12%|█▏        | 59/500 [00:25<02:47,  2.64it/s]

pred ==>  0


 12%|█▏        | 60/500 [00:25<02:55,  2.50it/s]

pred ==>  0


 12%|█▏        | 61/500 [00:26<03:01,  2.42it/s]

pred ==>  0


 12%|█▏        | 62/500 [00:26<02:44,  2.66it/s]

pred ==>  0


 13%|█▎        | 63/500 [00:27<03:23,  2.14it/s]

pred ==>  1


 13%|█▎        | 64/500 [00:27<03:20,  2.18it/s]

pred ==>  0


 13%|█▎        | 65/500 [00:28<03:18,  2.19it/s]

pred ==>  0


 13%|█▎        | 66/500 [00:28<02:52,  2.51it/s]

pred ==>  0


 13%|█▎        | 67/500 [00:28<02:57,  2.44it/s]

pred ==>  0


 14%|█▎        | 68/500 [00:29<03:01,  2.38it/s]

pred ==>  0


 14%|█▍        | 69/500 [00:29<03:02,  2.36it/s]

pred ==>  0


 14%|█▍        | 70/500 [00:30<03:05,  2.32it/s]

pred ==>  0


 14%|█▍        | 71/500 [00:30<03:04,  2.32it/s]

pred ==>  0


 14%|█▍        | 72/500 [00:31<03:10,  2.25it/s]

pred ==>  0


 15%|█▍        | 73/500 [00:31<02:54,  2.45it/s]

pred ==>  0


 15%|█▍        | 74/500 [00:31<02:44,  2.59it/s]

pred ==>  0


 15%|█▌        | 75/500 [00:32<02:56,  2.41it/s]

pred ==>  0


 15%|█▌        | 76/500 [00:32<03:06,  2.27it/s]

pred ==>  0


 15%|█▌        | 77/500 [00:33<03:11,  2.21it/s]

pred ==>  0


 16%|█▌        | 78/500 [00:33<02:58,  2.36it/s]

pred ==>  1


 16%|█▌        | 79/500 [00:34<03:05,  2.26it/s]

pred ==>  0


 16%|█▌        | 80/500 [00:34<02:48,  2.49it/s]

pred ==>  0


 16%|█▌        | 81/500 [00:34<02:54,  2.40it/s]

pred ==>  0


 16%|█▋        | 82/500 [00:35<02:38,  2.64it/s]

pred ==>  0


 17%|█▋        | 83/500 [00:35<02:49,  2.47it/s]

pred ==>  0


 17%|█▋        | 84/500 [00:36<02:56,  2.36it/s]

pred ==>  0


 17%|█▋        | 85/500 [00:36<03:03,  2.26it/s]

pred ==>  0


 17%|█▋        | 86/500 [00:37<03:04,  2.25it/s]

pred ==>  0


 17%|█▋        | 87/500 [00:37<03:11,  2.16it/s]

pred ==>  1


 18%|█▊        | 88/500 [00:37<03:11,  2.16it/s]

pred ==>  0


 18%|█▊        | 89/500 [00:38<03:18,  2.07it/s]

pred ==>  0


 18%|█▊        | 90/500 [00:39<03:41,  1.85it/s]

pred ==>  0


 18%|█▊        | 91/500 [00:39<03:33,  1.91it/s]

pred ==>  0


 18%|█▊        | 92/500 [00:40<03:10,  2.14it/s]

pred ==>  0


 19%|█▊        | 93/500 [00:40<03:10,  2.14it/s]

pred ==>  0


 19%|█▉        | 94/500 [00:40<02:47,  2.42it/s]

pred ==>  0


 19%|█▉        | 95/500 [00:41<02:31,  2.67it/s]

pred ==>  0


 19%|█▉        | 96/500 [00:41<02:41,  2.50it/s]

pred ==>  0


 19%|█▉        | 97/500 [00:41<02:46,  2.42it/s]

pred ==>  0


 20%|█▉        | 98/500 [00:42<03:18,  2.03it/s]

pred ==>  1


 20%|█▉        | 99/500 [00:43<03:13,  2.08it/s]

pred ==>  0


 20%|██        | 100/500 [00:43<03:10,  2.10it/s]

pred ==>  0


 20%|██        | 101/500 [00:43<03:04,  2.16it/s]

pred ==>  1


 20%|██        | 102/500 [00:44<03:00,  2.20it/s]

pred ==>  0


 21%|██        | 103/500 [00:44<03:04,  2.15it/s]

pred ==>  0


 21%|██        | 104/500 [00:45<02:45,  2.39it/s]

pred ==>  0


 21%|██        | 105/500 [00:45<02:53,  2.28it/s]

pred ==>  0


 21%|██        | 106/500 [00:46<02:57,  2.22it/s]

pred ==>  0


 21%|██▏       | 107/500 [00:46<03:09,  2.07it/s]

pred ==>  0


 22%|██▏       | 108/500 [00:47<02:50,  2.29it/s]

pred ==>  0


 22%|██▏       | 109/500 [00:47<02:51,  2.28it/s]

pred ==>  0


 22%|██▏       | 110/500 [00:47<02:33,  2.55it/s]

pred ==>  0


 22%|██▏       | 111/500 [00:48<02:38,  2.46it/s]

pred ==>  0


 22%|██▏       | 112/500 [00:48<02:40,  2.41it/s]

pred ==>  0


 23%|██▎       | 113/500 [00:48<02:26,  2.63it/s]

pred ==>  0


 23%|██▎       | 114/500 [00:49<02:34,  2.50it/s]

pred ==>  0


 23%|██▎       | 115/500 [00:49<02:39,  2.42it/s]

pred ==>  0


 23%|██▎       | 116/500 [00:50<02:25,  2.65it/s]

pred ==>  0


 23%|██▎       | 117/500 [00:50<02:17,  2.78it/s]

pred ==>  0


 24%|██▎       | 118/500 [00:51<03:13,  1.97it/s]

pred ==>  1


 24%|██▍       | 119/500 [00:51<03:06,  2.05it/s]

pred ==>  0


 24%|██▍       | 120/500 [00:52<03:00,  2.11it/s]

pred ==>  1


 24%|██▍       | 121/500 [00:52<02:58,  2.13it/s]

pred ==>  0


 24%|██▍       | 122/500 [00:53<02:53,  2.18it/s]

pred ==>  0


 25%|██▍       | 123/500 [00:53<02:53,  2.17it/s]

pred ==>  0


 25%|██▍       | 124/500 [00:53<02:50,  2.21it/s]

pred ==>  0


 25%|██▌       | 125/500 [00:54<02:51,  2.19it/s]

pred ==>  0


 25%|██▌       | 126/500 [00:54<02:29,  2.49it/s]

pred ==>  0


 25%|██▌       | 127/500 [00:55<02:15,  2.76it/s]

pred ==>  1


 26%|██▌       | 128/500 [00:55<02:05,  2.95it/s]

pred ==>  0


 26%|██▌       | 129/500 [00:55<02:19,  2.66it/s]

pred ==>  1


 26%|██▌       | 130/500 [00:56<02:25,  2.54it/s]

pred ==>  0


 26%|██▌       | 131/500 [00:56<02:30,  2.45it/s]

pred ==>  0


 26%|██▋       | 132/500 [00:57<02:33,  2.39it/s]

pred ==>  0


 27%|██▋       | 133/500 [00:57<02:34,  2.37it/s]

pred ==>  0


 27%|██▋       | 134/500 [00:57<02:17,  2.67it/s]

pred ==>  0


 27%|██▋       | 135/500 [00:58<02:25,  2.51it/s]

pred ==>  0


 27%|██▋       | 136/500 [00:58<02:36,  2.33it/s]

pred ==>  0


 27%|██▋       | 137/500 [00:59<02:28,  2.44it/s]

pred ==>  0


 28%|██▊       | 138/500 [00:59<02:41,  2.24it/s]

pred ==>  1


 28%|██▊       | 139/500 [01:00<02:40,  2.24it/s]

pred ==>  1


 28%|██▊       | 140/500 [01:00<02:22,  2.52it/s]

pred ==>  0


 28%|██▊       | 141/500 [01:00<02:15,  2.65it/s]

pred ==>  0


 28%|██▊       | 142/500 [01:01<02:10,  2.73it/s]

pred ==>  0


 29%|██▊       | 143/500 [01:01<02:21,  2.52it/s]

pred ==>  0


 29%|██▉       | 144/500 [01:01<02:26,  2.44it/s]

pred ==>  0


 29%|██▉       | 145/500 [01:02<02:13,  2.66it/s]

pred ==>  0


 29%|██▉       | 146/500 [01:02<02:03,  2.86it/s]

pred ==>  0


 29%|██▉       | 147/500 [01:02<02:12,  2.66it/s]

pred ==>  0


 30%|██▉       | 148/500 [01:03<02:19,  2.52it/s]

pred ==>  0


 30%|██▉       | 149/500 [01:03<02:23,  2.45it/s]

pred ==>  0


 30%|███       | 150/500 [01:04<02:26,  2.39it/s]

pred ==>  0


 30%|███       | 151/500 [01:04<02:10,  2.68it/s]

pred ==>  0


 30%|███       | 152/500 [01:05<02:19,  2.50it/s]

pred ==>  0


 31%|███       | 153/500 [01:05<02:06,  2.75it/s]

pred ==>  0


 31%|███       | 154/500 [01:05<02:13,  2.59it/s]

pred ==>  1


 31%|███       | 155/500 [01:06<02:29,  2.31it/s]

pred ==>  0


 31%|███       | 156/500 [01:06<02:16,  2.52it/s]

pred ==>  0


 31%|███▏      | 157/500 [01:07<02:29,  2.29it/s]

pred ==>  0


 32%|███▏      | 158/500 [01:07<02:29,  2.29it/s]

pred ==>  0


 32%|███▏      | 159/500 [01:08<02:49,  2.02it/s]

pred ==>  0


 32%|███▏      | 160/500 [01:08<03:18,  1.72it/s]

pred ==>  0


 32%|███▏      | 161/500 [01:09<03:02,  1.86it/s]

pred ==>  0


 32%|███▏      | 162/500 [01:09<02:51,  1.97it/s]

pred ==>  0


 33%|███▎      | 163/500 [01:10<02:44,  2.05it/s]

pred ==>  0


 33%|███▎      | 164/500 [01:10<02:38,  2.12it/s]

pred ==>  0


 33%|███▎      | 165/500 [01:11<02:33,  2.19it/s]

pred ==>  0


 33%|███▎      | 166/500 [01:11<02:33,  2.17it/s]

pred ==>  0


 33%|███▎      | 167/500 [01:12<02:37,  2.12it/s]

pred ==>  1


 34%|███▎      | 168/500 [01:12<02:34,  2.15it/s]

pred ==>  0


 34%|███▍      | 169/500 [01:13<02:37,  2.11it/s]

pred ==>  0


 34%|███▍      | 170/500 [01:13<02:33,  2.15it/s]

pred ==>  0


 34%|███▍      | 171/500 [01:14<02:59,  1.83it/s]

pred ==>  1


 34%|███▍      | 172/500 [01:14<02:32,  2.15it/s]

pred ==>  0


 35%|███▍      | 173/500 [01:14<02:30,  2.17it/s]

pred ==>  0


 35%|███▍      | 174/500 [01:15<02:29,  2.18it/s]

pred ==>  0


 35%|███▌      | 175/500 [01:15<02:28,  2.19it/s]

pred ==>  0


 35%|███▌      | 176/500 [01:16<02:29,  2.17it/s]

pred ==>  0


 35%|███▌      | 177/500 [01:16<02:27,  2.19it/s]

pred ==>  0


 36%|███▌      | 178/500 [01:17<02:09,  2.48it/s]

pred ==>  1


 36%|███▌      | 179/500 [01:17<02:15,  2.37it/s]

pred ==>  0


 36%|███▌      | 180/500 [01:17<02:02,  2.62it/s]

pred ==>  1


 36%|███▌      | 181/500 [01:18<02:08,  2.48it/s]

pred ==>  0


 36%|███▋      | 182/500 [01:18<01:57,  2.71it/s]

pred ==>  0


 37%|███▋      | 183/500 [01:18<01:46,  2.96it/s]

pred ==>  0


 37%|███▋      | 184/500 [01:19<01:56,  2.71it/s]

pred ==>  0


 37%|███▋      | 185/500 [01:19<02:06,  2.48it/s]

pred ==>  0


 37%|███▋      | 186/500 [01:20<02:10,  2.41it/s]

pred ==>  1


 37%|███▋      | 187/500 [01:20<01:57,  2.66it/s]

pred ==>  0


 38%|███▊      | 188/500 [01:20<01:48,  2.88it/s]

pred ==>  0


 38%|███▊      | 189/500 [01:21<01:41,  3.06it/s]

pred ==>  1


 38%|███▊      | 190/500 [01:21<01:35,  3.26it/s]

pred ==>  0


 38%|███▊      | 191/500 [01:21<01:49,  2.83it/s]

pred ==>  0


 38%|███▊      | 192/500 [01:22<01:59,  2.58it/s]

pred ==>  0


 39%|███▊      | 193/500 [01:22<02:04,  2.46it/s]

pred ==>  0


 39%|███▉      | 194/500 [01:22<01:51,  2.73it/s]

pred ==>  0


 39%|███▉      | 195/500 [01:23<02:04,  2.46it/s]

pred ==>  1


 39%|███▉      | 196/500 [01:23<01:53,  2.68it/s]

pred ==>  0


 39%|███▉      | 197/500 [01:24<02:02,  2.47it/s]

pred ==>  0


 40%|███▉      | 198/500 [01:24<02:08,  2.34it/s]

pred ==>  0


 40%|███▉      | 199/500 [01:25<02:20,  2.14it/s]

pred ==>  1


 40%|████      | 200/500 [01:25<02:21,  2.11it/s]

pred ==>  0


 40%|████      | 201/500 [01:26<02:28,  2.01it/s]

pred ==>  0


 40%|████      | 202/500 [01:26<02:26,  2.03it/s]

pred ==>  0


 41%|████      | 203/500 [01:27<02:26,  2.03it/s]

pred ==>  0


 41%|████      | 204/500 [01:27<02:28,  1.99it/s]

pred ==>  0


 41%|████      | 205/500 [01:28<02:24,  2.04it/s]

pred ==>  0


 41%|████      | 206/500 [01:28<02:22,  2.06it/s]

pred ==>  1


 41%|████▏     | 207/500 [01:29<02:04,  2.36it/s]

pred ==>  0


 42%|████▏     | 208/500 [01:29<02:08,  2.28it/s]

pred ==>  0


 42%|████▏     | 209/500 [01:29<01:53,  2.56it/s]

pred ==>  0


 42%|████▏     | 210/500 [01:30<01:57,  2.48it/s]

pred ==>  0


 42%|████▏     | 211/500 [01:30<02:00,  2.39it/s]

pred ==>  0


 42%|████▏     | 212/500 [01:31<02:04,  2.31it/s]

pred ==>  0


 43%|████▎     | 213/500 [01:31<02:04,  2.31it/s]

pred ==>  0


 43%|████▎     | 214/500 [01:31<02:04,  2.29it/s]

pred ==>  0


 43%|████▎     | 215/500 [01:32<02:04,  2.29it/s]

pred ==>  0


 43%|████▎     | 216/500 [01:32<02:05,  2.26it/s]

pred ==>  0


 43%|████▎     | 217/500 [01:33<02:04,  2.28it/s]

pred ==>  1


 44%|████▎     | 218/500 [01:33<02:03,  2.29it/s]

pred ==>  0


 44%|████▍     | 219/500 [01:34<02:02,  2.29it/s]

pred ==>  0


 44%|████▍     | 220/500 [01:34<01:49,  2.56it/s]

pred ==>  0


 44%|████▍     | 221/500 [01:34<01:54,  2.43it/s]

pred ==>  0


 44%|████▍     | 222/500 [01:35<01:55,  2.40it/s]

pred ==>  0


 45%|████▍     | 223/500 [01:35<01:43,  2.66it/s]

pred ==>  0


 45%|████▍     | 224/500 [01:36<01:49,  2.52it/s]

pred ==>  0


 45%|████▌     | 225/500 [01:36<01:51,  2.46it/s]

pred ==>  0


 45%|████▌     | 226/500 [01:36<01:52,  2.43it/s]

pred ==>  0


 45%|████▌     | 227/500 [01:37<01:53,  2.40it/s]

pred ==>  0


 46%|████▌     | 228/500 [01:37<01:54,  2.37it/s]

pred ==>  0


 46%|████▌     | 229/500 [01:38<01:43,  2.63it/s]

pred ==>  0


 46%|████▌     | 230/500 [01:38<01:48,  2.50it/s]

pred ==>  0


 46%|████▌     | 231/500 [01:38<01:51,  2.40it/s]

pred ==>  1


 46%|████▋     | 232/500 [01:39<01:53,  2.36it/s]

pred ==>  0


 47%|████▋     | 233/500 [01:39<01:41,  2.62it/s]

pred ==>  0


 47%|████▋     | 234/500 [01:40<01:46,  2.50it/s]

pred ==>  0


 47%|████▋     | 235/500 [01:40<01:49,  2.42it/s]

pred ==>  0


 47%|████▋     | 236/500 [01:40<01:38,  2.68it/s]

pred ==>  0


 47%|████▋     | 237/500 [01:41<01:43,  2.55it/s]

pred ==>  0


 48%|████▊     | 238/500 [01:41<01:45,  2.48it/s]

pred ==>  1


 48%|████▊     | 239/500 [01:42<01:47,  2.43it/s]

pred ==>  0


 48%|████▊     | 240/500 [01:42<01:48,  2.39it/s]

pred ==>  0


 48%|████▊     | 241/500 [01:42<01:37,  2.65it/s]

pred ==>  0


 48%|████▊     | 242/500 [01:43<01:42,  2.51it/s]

pred ==>  0


 49%|████▊     | 243/500 [01:43<01:31,  2.81it/s]

pred ==>  0


 49%|████▉     | 244/500 [01:44<01:54,  2.24it/s]

pred ==>  0


 49%|████▉     | 245/500 [01:44<01:53,  2.25it/s]

pred ==>  0


 49%|████▉     | 246/500 [01:44<01:40,  2.53it/s]

pred ==>  0


 49%|████▉     | 247/500 [01:45<01:44,  2.43it/s]

pred ==>  0


 50%|████▉     | 248/500 [01:45<01:32,  2.73it/s]

pred ==>  0


 50%|████▉     | 249/500 [01:45<01:25,  2.93it/s]

pred ==>  0


 50%|█████     | 250/500 [01:46<01:21,  3.08it/s]

pred ==>  0


 50%|█████     | 251/500 [01:46<01:29,  2.79it/s]

pred ==>  0


 50%|█████     | 252/500 [01:46<01:21,  3.05it/s]

pred ==>  0


 51%|█████     | 253/500 [01:47<01:28,  2.78it/s]

pred ==>  0


 51%|█████     | 254/500 [01:47<01:35,  2.59it/s]

pred ==>  0


 51%|█████     | 255/500 [01:48<01:27,  2.81it/s]

pred ==>  0


 51%|█████     | 256/500 [01:48<01:33,  2.61it/s]

pred ==>  0


 51%|█████▏    | 257/500 [01:49<01:53,  2.13it/s]

pred ==>  0


 52%|█████▏    | 258/500 [01:49<01:51,  2.18it/s]

pred ==>  0


 52%|█████▏    | 259/500 [01:50<01:54,  2.11it/s]

pred ==>  0


 52%|█████▏    | 260/500 [01:50<01:52,  2.13it/s]

pred ==>  0


 52%|█████▏    | 261/500 [01:51<01:51,  2.15it/s]

pred ==>  0


 52%|█████▏    | 262/500 [01:51<01:37,  2.45it/s]

pred ==>  0


 53%|█████▎    | 263/500 [01:51<01:39,  2.38it/s]

pred ==>  0


 53%|█████▎    | 264/500 [01:52<01:40,  2.36it/s]

pred ==>  0


 53%|█████▎    | 265/500 [01:52<01:40,  2.33it/s]

pred ==>  0


 53%|█████▎    | 266/500 [01:53<01:40,  2.32it/s]

pred ==>  0


 53%|█████▎    | 267/500 [01:53<01:41,  2.31it/s]

pred ==>  1


 54%|█████▎    | 268/500 [01:53<01:40,  2.30it/s]

pred ==>  0


 54%|█████▍    | 269/500 [01:54<01:40,  2.29it/s]

pred ==>  0


 54%|█████▍    | 270/500 [01:54<01:39,  2.31it/s]

pred ==>  0


 54%|█████▍    | 271/500 [01:55<01:39,  2.30it/s]

pred ==>  0


 54%|█████▍    | 272/500 [01:55<01:38,  2.30it/s]

pred ==>  0


 55%|█████▍    | 273/500 [01:56<01:28,  2.57it/s]

pred ==>  0


 55%|█████▍    | 274/500 [01:56<01:31,  2.47it/s]

pred ==>  0


 55%|█████▌    | 275/500 [01:56<01:34,  2.37it/s]

pred ==>  0


 55%|█████▌    | 276/500 [01:57<01:25,  2.63it/s]

pred ==>  0


 55%|█████▌    | 277/500 [01:57<01:28,  2.51it/s]

pred ==>  0


 56%|█████▌    | 278/500 [01:58<01:30,  2.45it/s]

pred ==>  0


 56%|█████▌    | 279/500 [01:58<01:31,  2.40it/s]

pred ==>  0


 56%|█████▌    | 280/500 [01:58<01:22,  2.66it/s]

pred ==>  0


 56%|█████▌    | 281/500 [01:59<01:14,  2.93it/s]

pred ==>  0


 56%|█████▋    | 282/500 [01:59<01:22,  2.64it/s]

pred ==>  1


 57%|█████▋    | 283/500 [01:59<01:15,  2.86it/s]

pred ==>  0


 57%|█████▋    | 284/500 [02:00<01:22,  2.63it/s]

pred ==>  0


 57%|█████▋    | 285/500 [02:00<01:13,  2.92it/s]

pred ==>  0


 57%|█████▋    | 286/500 [02:00<01:19,  2.68it/s]

pred ==>  0


 57%|█████▋    | 287/500 [02:01<01:24,  2.52it/s]

pred ==>  0


 58%|█████▊    | 288/500 [02:01<01:16,  2.76it/s]

pred ==>  0


 58%|█████▊    | 289/500 [02:01<01:11,  2.95it/s]

pred ==>  0


 58%|█████▊    | 290/500 [02:02<01:18,  2.66it/s]

pred ==>  0


 58%|█████▊    | 291/500 [02:02<01:23,  2.49it/s]

pred ==>  0


 58%|█████▊    | 292/500 [02:03<01:25,  2.43it/s]

pred ==>  0


 59%|█████▊    | 293/500 [02:03<01:27,  2.35it/s]

pred ==>  0


 59%|█████▉    | 294/500 [02:04<01:28,  2.32it/s]

pred ==>  0


 59%|█████▉    | 295/500 [02:04<01:29,  2.30it/s]

pred ==>  1


 59%|█████▉    | 296/500 [02:04<01:18,  2.59it/s]

pred ==>  0


 59%|█████▉    | 297/500 [02:05<01:23,  2.44it/s]

pred ==>  0


 60%|█████▉    | 298/500 [02:05<01:25,  2.37it/s]

pred ==>  0


 60%|█████▉    | 299/500 [02:06<01:26,  2.33it/s]

pred ==>  1


 60%|██████    | 300/500 [02:06<01:28,  2.26it/s]

pred ==>  0


 60%|██████    | 301/500 [02:07<01:27,  2.27it/s]

pred ==>  0


 60%|██████    | 302/500 [02:07<01:45,  1.87it/s]

pred ==>  1


 61%|██████    | 303/500 [02:08<01:39,  1.98it/s]

pred ==>  0


 61%|██████    | 304/500 [02:08<01:36,  2.03it/s]

pred ==>  0


 61%|██████    | 305/500 [02:09<01:23,  2.34it/s]

pred ==>  0


 61%|██████    | 306/500 [02:09<01:26,  2.25it/s]

pred ==>  0


 61%|██████▏   | 307/500 [02:10<01:26,  2.23it/s]

pred ==>  0


 62%|██████▏   | 308/500 [02:10<01:15,  2.54it/s]

pred ==>  1


 62%|██████▏   | 309/500 [02:10<01:18,  2.44it/s]

pred ==>  0


 62%|██████▏   | 310/500 [02:11<01:08,  2.75it/s]

pred ==>  0


 62%|██████▏   | 311/500 [02:11<01:12,  2.61it/s]

pred ==>  0


 62%|██████▏   | 312/500 [02:11<01:15,  2.49it/s]

pred ==>  0


 63%|██████▎   | 313/500 [02:12<01:08,  2.73it/s]

pred ==>  0


 63%|██████▎   | 314/500 [02:12<01:12,  2.55it/s]

pred ==>  0


 63%|██████▎   | 315/500 [02:12<01:06,  2.80it/s]

pred ==>  0


 63%|██████▎   | 316/500 [02:13<01:10,  2.61it/s]

pred ==>  1


 63%|██████▎   | 317/500 [02:13<01:04,  2.86it/s]

pred ==>  0


 64%|██████▎   | 318/500 [02:14<01:09,  2.62it/s]

pred ==>  0


 64%|██████▍   | 319/500 [02:14<01:13,  2.48it/s]

pred ==>  0


 64%|██████▍   | 320/500 [02:15<01:16,  2.37it/s]

pred ==>  1


 64%|██████▍   | 321/500 [02:15<01:16,  2.34it/s]

pred ==>  0


 64%|██████▍   | 322/500 [02:15<01:17,  2.30it/s]

pred ==>  0


 65%|██████▍   | 323/500 [02:16<01:08,  2.58it/s]

pred ==>  0


 65%|██████▍   | 324/500 [02:16<01:11,  2.47it/s]

pred ==>  1


 65%|██████▌   | 325/500 [02:17<01:14,  2.35it/s]

pred ==>  0


 65%|██████▌   | 326/500 [02:17<01:14,  2.33it/s]

pred ==>  0


 65%|██████▌   | 327/500 [02:18<01:22,  2.11it/s]

pred ==>  1


 66%|██████▌   | 328/500 [02:18<01:27,  1.97it/s]

pred ==>  0


 66%|██████▌   | 329/500 [02:19<01:22,  2.06it/s]

pred ==>  1


 66%|██████▌   | 330/500 [02:19<01:25,  1.99it/s]

pred ==>  0


 66%|██████▌   | 331/500 [02:20<01:21,  2.08it/s]

pred ==>  0


 66%|██████▋   | 332/500 [02:20<01:11,  2.35it/s]

pred ==>  0


 67%|██████▋   | 333/500 [02:20<01:03,  2.62it/s]

pred ==>  0


 67%|██████▋   | 334/500 [02:21<01:08,  2.43it/s]

pred ==>  0


 67%|██████▋   | 335/500 [02:21<01:09,  2.36it/s]

pred ==>  0


 67%|██████▋   | 336/500 [02:22<01:10,  2.31it/s]

pred ==>  0


 67%|██████▋   | 337/500 [02:22<01:10,  2.30it/s]

pred ==>  0


 68%|██████▊   | 338/500 [02:22<01:11,  2.26it/s]

pred ==>  0


 68%|██████▊   | 339/500 [02:23<01:11,  2.24it/s]

pred ==>  0


 68%|██████▊   | 340/500 [02:23<01:12,  2.21it/s]

pred ==>  1


 68%|██████▊   | 341/500 [02:24<01:11,  2.22it/s]

pred ==>  0


 68%|██████▊   | 342/500 [02:24<01:11,  2.21it/s]

pred ==>  0


 69%|██████▊   | 343/500 [02:25<01:04,  2.43it/s]

pred ==>  0


 69%|██████▉   | 344/500 [02:25<00:59,  2.60it/s]

pred ==>  0


 69%|██████▉   | 345/500 [02:25<01:02,  2.49it/s]

pred ==>  0


 69%|██████▉   | 346/500 [02:26<01:04,  2.40it/s]

pred ==>  0


 69%|██████▉   | 347/500 [02:26<01:04,  2.37it/s]

pred ==>  0


 70%|██████▉   | 348/500 [02:27<00:57,  2.63it/s]

pred ==>  0


 70%|██████▉   | 349/500 [02:27<01:00,  2.50it/s]

pred ==>  1


 70%|███████   | 350/500 [02:27<00:54,  2.76it/s]

pred ==>  0


 70%|███████   | 351/500 [02:28<00:50,  2.96it/s]

pred ==>  0


 70%|███████   | 352/500 [02:28<00:47,  3.10it/s]

pred ==>  0


 71%|███████   | 353/500 [02:28<00:52,  2.80it/s]

pred ==>  0


 71%|███████   | 354/500 [02:29<00:55,  2.61it/s]

pred ==>  1


 71%|███████   | 355/500 [02:29<00:50,  2.86it/s]

pred ==>  0


 71%|███████   | 356/500 [02:29<00:46,  3.11it/s]

pred ==>  0


 71%|███████▏  | 357/500 [02:30<00:51,  2.78it/s]

pred ==>  0


 72%|███████▏  | 358/500 [02:30<00:47,  2.97it/s]

pred ==>  0


 72%|███████▏  | 359/500 [02:30<00:44,  3.18it/s]

pred ==>  0


 72%|███████▏  | 360/500 [02:31<00:49,  2.84it/s]

pred ==>  0


 72%|███████▏  | 361/500 [02:31<00:45,  3.03it/s]

pred ==>  0


 72%|███████▏  | 362/500 [02:31<00:43,  3.16it/s]

pred ==>  0


 73%|███████▎  | 363/500 [02:32<00:48,  2.81it/s]

pred ==>  0


 73%|███████▎  | 364/500 [02:32<00:45,  3.01it/s]

pred ==>  0


 73%|███████▎  | 365/500 [02:32<00:49,  2.74it/s]

pred ==>  0


 73%|███████▎  | 366/500 [02:33<00:53,  2.51it/s]

pred ==>  1


 73%|███████▎  | 367/500 [02:33<00:54,  2.44it/s]

pred ==>  0


 74%|███████▎  | 368/500 [02:34<00:48,  2.73it/s]

pred ==>  0


 74%|███████▍  | 369/500 [02:34<00:50,  2.59it/s]

pred ==>  0


 74%|███████▍  | 370/500 [02:34<00:53,  2.43it/s]

pred ==>  0


 74%|███████▍  | 371/500 [02:35<00:47,  2.71it/s]

pred ==>  0


 74%|███████▍  | 372/500 [02:35<00:49,  2.58it/s]

pred ==>  0


 75%|███████▍  | 373/500 [02:36<00:53,  2.39it/s]

pred ==>  0


 75%|███████▍  | 374/500 [02:36<00:53,  2.34it/s]

pred ==>  0


 75%|███████▌  | 375/500 [02:37<00:53,  2.36it/s]

pred ==>  1


 75%|███████▌  | 376/500 [02:37<00:47,  2.63it/s]

pred ==>  0


 75%|███████▌  | 377/500 [02:37<00:48,  2.55it/s]

pred ==>  0


 76%|███████▌  | 378/500 [02:38<00:50,  2.43it/s]

pred ==>  0


 76%|███████▌  | 379/500 [02:38<00:51,  2.36it/s]

pred ==>  0


 76%|███████▌  | 380/500 [02:39<00:51,  2.35it/s]

pred ==>  0


 76%|███████▌  | 381/500 [02:39<00:51,  2.30it/s]

pred ==>  1


 76%|███████▋  | 382/500 [02:40<00:51,  2.27it/s]

pred ==>  0


 77%|███████▋  | 383/500 [02:40<00:51,  2.29it/s]

pred ==>  0


 77%|███████▋  | 384/500 [02:40<00:50,  2.30it/s]

pred ==>  1


 77%|███████▋  | 385/500 [02:41<00:49,  2.30it/s]

pred ==>  0


 77%|███████▋  | 386/500 [02:41<00:49,  2.31it/s]

pred ==>  1


 77%|███████▋  | 387/500 [02:42<00:48,  2.31it/s]

pred ==>  0


 78%|███████▊  | 388/500 [02:42<00:48,  2.31it/s]

pred ==>  0


 78%|███████▊  | 389/500 [02:42<00:42,  2.58it/s]

pred ==>  0


 78%|███████▊  | 390/500 [02:43<00:38,  2.87it/s]

pred ==>  0


 78%|███████▊  | 391/500 [02:43<00:35,  3.07it/s]

pred ==>  0


 78%|███████▊  | 392/500 [02:43<00:38,  2.79it/s]

pred ==>  1


 79%|███████▊  | 393/500 [02:44<00:40,  2.64it/s]

pred ==>  0


 79%|███████▉  | 394/500 [02:44<00:42,  2.47it/s]

pred ==>  0


 79%|███████▉  | 395/500 [02:44<00:38,  2.75it/s]

pred ==>  0


 79%|███████▉  | 396/500 [02:45<00:40,  2.56it/s]

pred ==>  0


 79%|███████▉  | 397/500 [02:45<00:36,  2.79it/s]

pred ==>  1


 80%|███████▉  | 398/500 [02:46<00:38,  2.63it/s]

pred ==>  0


 80%|███████▉  | 399/500 [02:46<00:40,  2.51it/s]

pred ==>  1


 80%|████████  | 400/500 [02:47<00:41,  2.42it/s]

pred ==>  0


 80%|████████  | 401/500 [02:47<00:36,  2.68it/s]

pred ==>  0


 80%|████████  | 402/500 [02:47<00:33,  2.91it/s]

pred ==>  0


 81%|████████  | 403/500 [02:47<00:31,  3.07it/s]

pred ==>  0


 81%|████████  | 404/500 [02:48<00:34,  2.80it/s]

pred ==>  0


 81%|████████  | 405/500 [02:48<00:31,  3.04it/s]

pred ==>  0


 81%|████████  | 406/500 [02:49<00:33,  2.77it/s]

pred ==>  0


 81%|████████▏ | 407/500 [02:49<00:35,  2.61it/s]

pred ==>  0


 82%|████████▏ | 408/500 [02:49<00:32,  2.84it/s]

pred ==>  1


 82%|████████▏ | 409/500 [02:50<00:33,  2.73it/s]

pred ==>  0


 82%|████████▏ | 410/500 [02:50<00:37,  2.38it/s]

pred ==>  1


 82%|████████▏ | 411/500 [02:51<00:37,  2.36it/s]

pred ==>  0


 82%|████████▏ | 412/500 [02:51<00:33,  2.62it/s]

pred ==>  0


 83%|████████▎ | 413/500 [02:51<00:35,  2.47it/s]

pred ==>  0


 83%|████████▎ | 414/500 [02:52<00:35,  2.42it/s]

pred ==>  0


 83%|████████▎ | 415/500 [02:52<00:35,  2.40it/s]

pred ==>  0


 83%|████████▎ | 416/500 [02:52<00:31,  2.65it/s]

pred ==>  0


 83%|████████▎ | 417/500 [02:53<00:32,  2.54it/s]

pred ==>  1


 84%|████████▎ | 418/500 [02:53<00:29,  2.78it/s]

pred ==>  0


 84%|████████▍ | 419/500 [02:54<00:31,  2.56it/s]

pred ==>  0


 84%|████████▍ | 420/500 [02:54<00:32,  2.48it/s]

pred ==>  0


 84%|████████▍ | 421/500 [02:54<00:28,  2.77it/s]

pred ==>  0


 84%|████████▍ | 422/500 [02:55<00:26,  2.96it/s]

pred ==>  0


 85%|████████▍ | 423/500 [02:55<00:28,  2.71it/s]

pred ==>  0


 85%|████████▍ | 424/500 [02:56<00:29,  2.59it/s]

pred ==>  0


 85%|████████▌ | 425/500 [02:56<00:35,  2.12it/s]

pred ==>  1


 85%|████████▌ | 426/500 [02:57<00:34,  2.15it/s]

pred ==>  0


 85%|████████▌ | 427/500 [02:57<00:40,  1.81it/s]

pred ==>  1


 86%|████████▌ | 428/500 [02:58<00:33,  2.15it/s]

pred ==>  0


 86%|████████▌ | 429/500 [02:58<00:32,  2.18it/s]

pred ==>  0


 86%|████████▌ | 430/500 [02:59<00:31,  2.20it/s]

pred ==>  0


 86%|████████▌ | 431/500 [02:59<00:30,  2.23it/s]

pred ==>  0


 86%|████████▋ | 432/500 [02:59<00:30,  2.25it/s]

pred ==>  0


 87%|████████▋ | 433/500 [03:00<00:29,  2.27it/s]

pred ==>  0


 87%|████████▋ | 434/500 [03:00<00:29,  2.28it/s]

pred ==>  0


 87%|████████▋ | 435/500 [03:01<00:25,  2.57it/s]

pred ==>  0


 87%|████████▋ | 436/500 [03:01<00:22,  2.86it/s]

pred ==>  0


 87%|████████▋ | 437/500 [03:01<00:20,  3.04it/s]

pred ==>  0


 88%|████████▊ | 438/500 [03:02<00:22,  2.78it/s]

pred ==>  0


 88%|████████▊ | 439/500 [03:02<00:20,  2.97it/s]

pred ==>  0


 88%|████████▊ | 440/500 [03:02<00:19,  3.16it/s]

pred ==>  0


 88%|████████▊ | 441/500 [03:03<00:20,  2.83it/s]

pred ==>  0


 88%|████████▊ | 442/500 [03:03<00:21,  2.66it/s]

pred ==>  0


 89%|████████▊ | 443/500 [03:03<00:22,  2.50it/s]

pred ==>  0


 89%|████████▉ | 444/500 [03:04<00:20,  2.79it/s]

pred ==>  0


 89%|████████▉ | 445/500 [03:04<00:21,  2.57it/s]

pred ==>  0


 89%|████████▉ | 446/500 [03:04<00:19,  2.81it/s]

pred ==>  0


 89%|████████▉ | 447/500 [03:05<00:17,  3.05it/s]

pred ==>  0


 90%|████████▉ | 448/500 [03:05<00:18,  2.78it/s]

pred ==>  0


 90%|████████▉ | 449/500 [03:06<00:20,  2.52it/s]

pred ==>  1


 90%|█████████ | 450/500 [03:06<00:17,  2.81it/s]

pred ==>  0


 90%|█████████ | 451/500 [03:06<00:19,  2.57it/s]

pred ==>  0


 90%|█████████ | 452/500 [03:07<00:19,  2.47it/s]

pred ==>  0


 91%|█████████ | 453/500 [03:07<00:22,  2.08it/s]

pred ==>  0


 91%|█████████ | 454/500 [03:08<00:19,  2.41it/s]

pred ==>  0


 91%|█████████ | 455/500 [03:08<00:19,  2.34it/s]

pred ==>  0


 91%|█████████ | 456/500 [03:09<00:19,  2.31it/s]

pred ==>  0


 91%|█████████▏| 457/500 [03:09<00:18,  2.30it/s]

pred ==>  1


 92%|█████████▏| 458/500 [03:09<00:18,  2.22it/s]

pred ==>  0


 92%|█████████▏| 459/500 [03:10<00:18,  2.22it/s]

pred ==>  0


 92%|█████████▏| 460/500 [03:10<00:16,  2.49it/s]

pred ==>  0


 92%|█████████▏| 461/500 [03:11<00:16,  2.37it/s]

pred ==>  1


 92%|█████████▏| 462/500 [03:11<00:16,  2.35it/s]

pred ==>  1


 93%|█████████▎| 463/500 [03:12<00:16,  2.23it/s]

pred ==>  0


 93%|█████████▎| 464/500 [03:12<00:16,  2.25it/s]

pred ==>  0


 93%|█████████▎| 465/500 [03:12<00:13,  2.53it/s]

pred ==>  0


 93%|█████████▎| 466/500 [03:13<00:14,  2.39it/s]

pred ==>  0


 93%|█████████▎| 467/500 [03:13<00:13,  2.36it/s]

pred ==>  0


 94%|█████████▎| 468/500 [03:14<00:12,  2.62it/s]

pred ==>  1


 94%|█████████▍| 469/500 [03:14<00:10,  2.85it/s]

pred ==>  0


 94%|█████████▍| 470/500 [03:14<00:09,  3.02it/s]

pred ==>  0


 94%|█████████▍| 471/500 [03:14<00:09,  3.21it/s]

pred ==>  0


 94%|█████████▍| 472/500 [03:15<00:09,  2.86it/s]

pred ==>  0


 95%|█████████▍| 473/500 [03:15<00:10,  2.68it/s]

pred ==>  0


 95%|█████████▍| 474/500 [03:16<00:09,  2.89it/s]

pred ==>  0


 95%|█████████▌| 475/500 [03:16<00:08,  3.06it/s]

pred ==>  0


 95%|█████████▌| 476/500 [03:16<00:08,  2.78it/s]

pred ==>  0


 95%|█████████▌| 477/500 [03:17<00:08,  2.56it/s]

pred ==>  1


 96%|█████████▌| 478/500 [03:17<00:08,  2.47it/s]

pred ==>  0


 96%|█████████▌| 479/500 [03:18<00:08,  2.37it/s]

pred ==>  0


 96%|█████████▌| 480/500 [03:18<00:08,  2.36it/s]

pred ==>  0


 96%|█████████▌| 481/500 [03:18<00:07,  2.61it/s]

pred ==>  0


 96%|█████████▋| 482/500 [03:19<00:07,  2.52it/s]

pred ==>  0


 97%|█████████▋| 483/500 [03:19<00:06,  2.76it/s]

pred ==>  0


 97%|█████████▋| 484/500 [03:19<00:06,  2.62it/s]

pred ==>  0


 97%|█████████▋| 485/500 [03:20<00:05,  2.51it/s]

pred ==>  0


 97%|█████████▋| 486/500 [03:20<00:05,  2.79it/s]

pred ==>  0


 97%|█████████▋| 487/500 [03:20<00:04,  3.04it/s]

pred ==>  0


 98%|█████████▊| 488/500 [03:21<00:03,  3.17it/s]

pred ==>  0


 98%|█████████▊| 489/500 [03:21<00:03,  2.83it/s]

pred ==>  0


 98%|█████████▊| 490/500 [03:22<00:03,  2.63it/s]

pred ==>  0


 98%|█████████▊| 491/500 [03:22<00:03,  2.46it/s]

pred ==>  1


 98%|█████████▊| 492/500 [03:22<00:02,  2.69it/s]

pred ==>  0


 99%|█████████▊| 493/500 [03:23<00:02,  2.56it/s]

pred ==>  0


 99%|█████████▉| 494/500 [03:23<00:02,  2.47it/s]

pred ==>  0


 99%|█████████▉| 495/500 [03:24<00:02,  2.42it/s]

pred ==>  0


 99%|█████████▉| 496/500 [03:24<00:01,  2.36it/s]

pred ==>  0


 99%|█████████▉| 497/500 [03:25<00:01,  2.32it/s]

pred ==>  0


100%|█████████▉| 498/500 [03:25<00:00,  2.64it/s]

pred ==>  0


100%|█████████▉| 499/500 [03:25<00:00,  2.42it/s]

pred ==>  0


100%|██████████| 500/500 [03:26<00:00,  2.42it/s]

pred ==>  0
151
n_corrects ==>  151


### 500개 뉴스 기사

결과 : 151 개

실제 pred 1 =>  60 개  (긍정)
실제 pred 0 =>  440 개  (나머지 부정)

어떤 의미인지???